### PR TITLE
Server packets

### PR DESCRIPTION
**Pull request by hicytree**

### CODE REVIEW

File: include/client_core.h [hunk 1]
```diff
@@ -30,9 +32,9 @@ class ClientCore {
     void render();                                  // Render the game state to the user
 
     bool connected;                // Connection state
-    Client client;                 // client.conn_sock = socket, 
+    Client client;                 // client.conn_sock = socket,
+    Packet packet;
     ServerData server_updates;
```
Comment (hgrehm) [inline include/client_core.h:37]: Why do we have packet as a member var? Is there a single persistent packet we want the ClientCore to remember, or can this be passed between functions?

File: include/packet.h [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+class Packet {
+    public:
+        struct ServerPacket {
+            int id;
+            std::vector<int> coor;
+            std::string message;
+        };
+
+        void serialize_server_packet(ServerPacket* pkt, char* buf);
+        void deserialize_server_packet(ServerPacket* pkt, char* buf);
```
Comment (hgrehm) [inline include/packet.h:13]: Could these be static? i.e., if we call using `somePacket.serialize(actualPacket, buf)`, how is the calling packet `somePacket` useful if this is not a static func?

File: include/client_core.h (hunk 1 above)
Comment (hicytree) [reply include/client_core.h]: fixed

File: include/packet.h (hunk 2 above)
Comment (hicytree) [reply include/packet.h]: forgot static functions were a thing tbh

**Review by hgrehm — APPROVED** [review]

**Review by hgrehm — APPROVED** [review]

**Pull request merged**